pressbooks/pressbooks
# Allow import/export of contributors list

Issue #1824 

This PR adds a new bulk action to export the list of contributors as a JSON file. It also adds a new form that allows users to import a list of contributors with all meta information.

This PR should also fix #2360.

#### How to test

1. Select a book that has contributors.
1. Go to Admin Book > Book Info > Contributors.
1. There should be a new form 'Import Contributors' and a new bulk action 'Download JSON'.
1. Select the list of contributors to export and download the JSON.
1. Check if the file has all contributors and all the metadata.
1. Go to another book's contributors list and import the previously download file.
1. Make sure that all contributors are imported and all metadata is present.

Things to notice: if a contributor with the same slug already exists, all the metadata will be updated (If not required, this could be removed).

#### TODO
- [ ] Define the 'Our Guide' URL that is displayed below the file input field.
- [ ] Improve message for invalid contributors.
